File: HrmsWebApiCore/WebApiCore/Models/SalaryProcess/BlockSalaryProcessModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using WebApiCore.ViewModels.SalaryProcess;

namespace WebApiCore.Models.SalaryProcess
{
    public class BlockSalaryProcessModel
    {
        public int PeriodID  {get;set;}
        public int TaxYearID {get;set;}
        public string PeriodName  {get;set;}
        public int YearID  {get;set;}
        public int Salaryhead { get; set; }
        public string EmployeeCode  {get;set;}
        public int CompanyID {get;set;}
        public int Grade  {get;set;}
        public int UserTypeID { get; set; }
        public int Bonustype { get; set; }
        public List<BlockSalaryProcessViewModel> BlockSalaryViewModel { get; set; }
    }
}

File: HrmsAngular/src/app/models/salary-process/block-salary-process.model.ts
import { BlockSalaryProcessView } from "./block-salary-process-view.model";

export class BlockSalaryProcess {
    periodID :number;
    structureID  :number;
    taxYearID :number;
    periodName  :string;
    yearID  :number;
    salaryhead:number;
    employeeCode :string;
    companyID :number;
    grade  :number;
    department:string;
    empCode:string;
    empName:string;
    designation:string;
    userTypeID:number;
    blockSalaryViewModel:BlockSalaryProcessView[]
    
}
